ale/drivers/cube_driver.py

```diff
@@ -0,0 +1,224 @@
+            if table['Name'] == 'InstrumentPointing':
+                self.inst_pointing_table = parse_rotation_table(table, field_data)
+            elif table['Name'] == 'BodyRotation':
+                self.body_orientation_table = parse_rotation_table(table, field_data)
+            elif table['Name'] == 'InstrumentPosition':
+                self.inst_position_table = parse_position_table(field_data)
+            elif table['Name'] == 'SunPosition':
+                self.sun_position_table = parse_position_table(field_data)
+
+    @property
+    def instrument_id(self):
+        return self.label['IsisCube']['Instrument']['InstrumentId']
+
+    @property
+    def start_time(self):
+        return self.label['IsisCube']['Instrument']['StartTime']
+
+    @property
+    def image_lines(self):
+        return self.label['IsisCube']['Core']['Dimensions']['Lines']
+
+    @property
+    def image_samples(self):
+        return self.label['IsisCube']['Core']['Dimensions']['Samples']
+
+    @property
+    def interpolation_method(self):
+        return 'hermite'
+
+    @property
+    def number_of_quaternions(self):
+        return len(self.sensor_orientation)
+
+    @property
+    def number_of_ephemerides(self):
+        return len(self.sensor_position)
+
+    @property
+    def target_name(self):
+        return self.label['IsisCube']['Instrument']['TargetName']
+
+    @property
+    def starting_ephemeris_time(self):
+        return self.inst_position_table['Times'][0]
+
+    @property
+    def ending_ephemeris_time(self):
+        return self.inst_position_table['Times'][-1]
+
+    @property
+    def detector_center(self):
+        return [
+            self.label['NaifKeywords']['INS{}_BORESIGHT_LINE'.format(self.ikid)],
+            self.label['NaifKeywords']['INS{}_BORESIGHT_SAMPLE'.format(self.ikid)]
+        ]
+
+    @property
+    def spacecraft_name(self):
+        return self.label['IsisCube']['Instrument']['SpacecraftName']
+
+    @property
+    def ikid(self):
+        return self.label['IsisCube']['Kernels']['NaifIkCode']
+
+    @property
+    def fikid(self):
+        pass
+
+    @property
+    def focal2pixel_lines(self):
+        return self.label['NaifKeywords']['INS{}_ITRANSL'.format(self.ikid)]
+
+    @property
+    def focal2pixel_samples(self):
+        return self.label['NaifKeywords']['INS{}_ITRANSS'.format(self.ikid)]
+
+    @property
+    def focal_length(self):
+        return self.label['NaifKeywords']['INS{}_FOCAL_LENGTH'.format(self.ikid)]
+
+    @property
+    def body_radii(self):
+        for key in self.label['NaifKeywords']:
+            if re.match('BODY-?\d*_RADII', key[0]):
+                return self.label['NaifKeywords'][key[0]]
+
+    @property
+    def semimajor(self):
+        return self.body_radii[0]
+
+    @property
+    def semiminor(self):
+        return self.body_radii[2]
+
+    @property
+    def reference_frame(self):
+        return self.body_orientation_table['TimeDependentFrames'][0]
```

Comment:
What about changing these to `self.appropriate_table.get('KEY', empty_default)`? I guess my real question is about error handling. The funcs above pack dicts `if all()` without any potential that `all() == False`. That is fine I think, but then this code needs to handle those instances. If only because things like the `number_of_ephemerides` properties will end up throwing a KeyError (which does not make sense unless you have seen this code and followed the execution path).